diff --git a/node_modules/@railgun-community/wallet/dist/services/poi/poi-node-request.js b/node_modules/@railgun-community/wallet/dist/services/poi/poi-node-request.js
index e49f922..5612634 100644
--- a/node_modules/@railgun-community/wallet/dist/services/poi/poi-node-request.js
+++ b/node_modules/@railgun-community/wallet/dist/services/poi/poi-node-request.js
@@ -68,7 +68,7 @@ class POINodeRequest {
     }
     async attemptRequestWithFallbacks(method, params, nodeUrlAttemptIndex = 0, finalAttempt = false) {
         try {
-            const url = this.getNodeURL(nodeUrlAttemptIndex);
+            const url = this.getNextNodeURL();
             const res = await (0, shared_models_1.promiseTimeout)(POINodeRequest.jsonRpcRequest(url, method, params), 60000);
             return res;
         }
